From 6beace8494fea0e2bfdc932467b1e0e5a8fb484b Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Fri, 7 Feb 2020 11:30:53 +0100
Subject: [PATCH 207/529] rtc: stm32: change PM callbacks to "_noirq()"

The rtc drivers stop the RTCAPB clock during suspend, but the
irq handle from RTC is called before starting clock. Then we are
blocked during IP access.
As the driver is registered as wakeup_source we can change PM
callbacks to '_nor_irq()' to resume automatically.

Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: Ia8d4f707afdb567ad38185643ed4ac3970a4d960
---
 drivers/rtc/rtc-stm32.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/drivers/rtc/rtc-stm32.c
+++ b/drivers/rtc/rtc-stm32.c
@@ -892,8 +892,9 @@ static int stm32_rtc_resume(struct devic
 }
 #endif
 
-static SIMPLE_DEV_PM_OPS(stm32_rtc_pm_ops,
-			 stm32_rtc_suspend, stm32_rtc_resume);
+static const struct dev_pm_ops stm32_rtc_pm_ops = {
+	SET_NOIRQ_SYSTEM_SLEEP_PM_OPS(stm32_rtc_suspend, stm32_rtc_resume)
+};
 
 static struct platform_driver stm32_rtc_driver = {
 	.probe		= stm32_rtc_probe,
